From 35f6b872b7bd4cd0245f4c8b2e4a2cdc86fe8151 Mon Sep 17 00:00:00 2001
From: Takuya Sakata <takuya.sakata.wz@bp.renesas.com>
Date: Tue, 16 Oct 2018 15:31:31 +0900
Subject: [PATCH] Update optee_os Rev1.0.16 rev2

 - Fix to set the initial value for a parameter in TEE_AEInit.
 - Fix to clear the read cache of standalone_fs_create.
---
 core/arch/arm/plat-rcar/tee/tee_standalone_fs.c | 4 ++++
 lib/libutee/tee_api_operations.c                | 1 +
 2 files changed, 5 insertions(+)

diff --git a/core/arch/arm/plat-rcar/tee/tee_standalone_fs.c b/core/arch/arm/plat-rcar/tee/tee_standalone_fs.c
index c51b2cd..a475940 100644
--- a/core/arch/arm/plat-rcar/tee/tee_standalone_fs.c
+++ b/core/arch/arm/plat-rcar/tee/tee_standalone_fs.c
@@ -942,6 +942,10 @@ static void spi_write_record_data(struct spif_record_info *record_info,
 	}
 
 	record_info->record_head.data_len = dpos;
+
+	if (g_record_data_rdesc != NULL) {
+		g_record_data_rdesc = NULL;
+	}
 }
 
 static void spi_get_parent_dir(const char *path, size_t path_len,
diff --git a/lib/libutee/tee_api_operations.c b/lib/libutee/tee_api_operations.c
index 2b11fd2..036ec13 100644
--- a/lib/libutee/tee_api_operations.c
+++ b/lib/libutee/tee_api_operations.c
@@ -1343,6 +1343,7 @@ TEE_Result TEE_AEInit(TEE_OperationHandle operation, const void *nonce,
 	if (res != TEE_SUCCESS)
 		goto out;
 
+	operation->buffer_offs = 0U;
 	operation->ae_tag_len = tagLen / 8;
 	operation->info.handleState |= TEE_HANDLE_FLAG_INITIALIZED;
 
-- 
1.9.1

